Clean up and expand .gitignore file

rye opened the pull request:
- Adds `/logs` to the `.gitignore`.
- Removes `#`-preceded lines after comments

Anyone have recommendations on stuff I can remove in addition to what I've done here? Worth a pass through just to see if anything here actually hits the ignores.

rye committed:
gitignore: Ignore logs files

rye committed:
gitignore: Remove strange #'s

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
gitignore: Remove a couple of exceptions

Suggested-by: Hawken Rives <hawkrives@fastmail.fm>
Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
gitignore: Add a preceding slash to docs/ exclusion

Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
Tested-by: Kristofer Rye <kristofer.rye@gmail.com>

hawkrives commented on .gitignore:
```diff
@@ -23,29 +21,21 @@ DerivedData
-buck-out/
-\.buckd/
-android/app/libs
+# Android Signing Keys
```
looking at this again, this section could move down and be joined up with the `# android keystores` section

rye replied:
Actually, the `*.keystore` rule is a duplicate.

hawkrives commented on .gitignore:
```diff
@@ -64,7 +54,7 @@ scripts/travisci_deploy_key
 fastlane/play-private-key.json
```
and this line could also move down to `# android keystores`, and we could probably remove the `travisci_deploy_key` line, too.

rye committed:
gitignore: Remove duplicated rule

This was already present below.

Suggested-by: Hawken Rives <hawkrives@fastmail.fm>
Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>

rye committed:
gitignore: Merge deploy key sections and remove travis deploy key

Suggested-by: Hawken Rives <hawkrives@fastmail.fm>
Signed-off-by: Kristofer Rye <kristofer.rye@gmail.com>
